src/apps/replication/meta_server/meta_state_service_simple.cpp

```diff
@@ -275,33 +283,33 @@ namespace dsn
             binary_writer log_writer;
             int magic = 0xdeadbeef;
             log_writer.write(magic);
-            log_writer.write(static_cast<size_t>(log_writer.total_size()));
-            log_writer.write(log_writer.get_buffer());
-            
+            log_writer.write(temperal_writer.get_buffer());
+
             size_t log_size = log_writer.total_size();
             _log_lock.lock();
             uint64_t log_offset = _offset;
             _offset += log_writer.total_size();
 
             auto returned_task_ptr = tasking::create_late_task(cb_code, cb_create, 0, tracker);
-            auto continuation_task = std::make_unique<operation>(false, [=](bool log_succeed)
+            auto continuation_task = std::unique_ptr<operation>(new operation(false, [=](bool log_succeed)
             {
                 dassert(log_succeed, "we cannot handle logging failure now");
                 __err_cb_bind_and_enqueue(returned_task_ptr, create_node_internal(node, value), 0);
-            });
+            }));
             auto continuation_task_ptr = continuation_task.get();
             _task_queue.emplace(std::move(continuation_task));
             _log_lock.unlock();
+            blob shared_blob = log_writer.get_buffer();
              file::write(
                 _log,
-                log_writer.get_buffer().buffer_ptr(),
+                shared_blob.buffer_ptr(),
                 log_writer.total_size(),
                 log_offset,
                 META_STATE_SERVICE_SIMPLE_INTERNAL_TASK,
```

Comment:
better to use name convention of the task code as LPC_META_STATE_SERVICE_SIMPLE_INTERNAL